charts/tidb-lightning/templates/scripts/_start_lightning.sh.tpl

```diff
@@ -16,18 +16,13 @@ fi
 /tidb-lightning \
     --pd-urls={{ .Values.targetTidbCluster.name }}-pd.{{ .Values.targetTidbCluster.namespace | default .Release.Namespace }}:2379 \
     --status-addr=0.0.0.0:8289 \
+{{- if eq .Values.backend "importer" }}
     --importer={{ .Values.targetTidbCluster.name }}-importer.{{ .Values.targetTidbCluster.namespace | default .Release.Namespace }}:8287 \
+{{- else if eq .Values.backend "tidb" }}
+    --backend tidb \
+{{- end }}
     --server-mode=false \
     --tidb-user={{ .Values.targetTidbCluster.user | default "root" }} \
     --tidb-host={{ .Values.targetTidbCluster.name }}-tidb.{{ .Values.targetTidbCluster.namespace | default .Release.Namespace }} \
     --d=${data_dir} \
     --config=/etc/tidb-lightning/tidb-lightning.toml
-
-if [ $? != 0 ]; then
```

Comment:
If we aren't checking the exit code then we need `set -e`. Best to have `set -euo pipefail`